src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.cs

```diff
@@ -42,29 +64,119 @@ public partial class HybridWebViewHandler : IHybridWebViewHandler
-        {
+		public static CommandMapper<IHybridWebView, IHybridWebViewHandler> CommandMapper = new(ViewCommandMapper)
+		{
 			[nameof(IHybridWebView.EvaluateJavaScriptAsync)] = MapEvaluateJavaScriptAsync,
+			[nameof(IHybridWebView.InvokeJavaScriptAsync)] = MapInvokeJavaScriptAsync,
 			[nameof(IHybridWebView.SendRawMessage)] = MapSendRawMessage,
-        };
+		};
+
+		public HybridWebViewHandler() : base(Mapper, CommandMapper)
+		{
+		}
+
+		public HybridWebViewHandler(IPropertyMapper? mapper = null, CommandMapper? commandMapper = null)
+			: base(mapper ?? Mapper, commandMapper ?? CommandMapper)
+		{
+		}
+
+		IHybridWebView IHybridWebViewHandler.VirtualView => VirtualView;
+
+		PlatformView IHybridWebViewHandler.PlatformView => PlatformView;
+
+
+
+		/// <summary>
+		/// Handler for when the an Async JavaScript task has completed and needs to notify .NET.
+		/// </summary>
+		private void AsyncTaskCompleted(string taskId, string result)
+		{
+			// Look for the callback in the list of pending callbacks
+			if (!string.IsNullOrEmpty(taskId) && _asyncTaskCallbacks.TryGetValue(taskId, out var callback))
+			{
+				// Get the callback and remove it from the list
+				callback.SetResult(result);
+
+				// Remove the callback
+				_asyncTaskCallbacks.TryRemove(taskId, out var _);
+			}
+		}
 
-        public HybridWebViewHandler() : base(Mapper, CommandMapper)
-        {
-        }
+		void MessageReceived(string rawMessage)
+		{
+			if (string.IsNullOrEmpty(rawMessage))
+			{
+				throw new ArgumentException($"The raw message cannot be null or empty.", nameof(rawMessage));
+			}
+#if !NETSTANDARD2_0
+			var indexOfPipe = rawMessage.IndexOf('|', StringComparison.Ordinal);
+#else
+			var indexOfPipe = rawMessage.IndexOf("|", StringComparison.Ordinal);
+#endif
+			if (indexOfPipe == -1)
+			{
+				throw new ArgumentException($"The raw message must contain a pipe character ('|').", nameof(rawMessage));
+			}
 
-        public HybridWebViewHandler(IPropertyMapper? mapper = null, CommandMapper? commandMapper = null)
-            : base(mapper ?? Mapper, commandMapper ?? CommandMapper)
-        {
-        }
+			var messageType = rawMessage.Substring(0, indexOfPipe);
+			var messageContent = rawMessage.Substring(indexOfPipe + 1);
 
-        IHybridWebView IHybridWebViewHandler.VirtualView => VirtualView;
+			switch (messageType)
+			{
+				case "InvokeMethodCompleted":
+					{
+						var sections = messageContent.Split('|');
+						var taskId = sections[0];
+						var result = sections[1];
+						AsyncTaskCompleted(taskId, result);
+					}
+					break;
+				case "RawMessage":
+					VirtualView?.RawMessageReceived(messageContent);
+					break;
+				default:
+					throw new ArgumentException($"The message type '{messageType}' is not recognized.", nameof(rawMessage));
+			}
+		}
+
+
+		public static async void MapInvokeJavaScriptAsync(IHybridWebViewHandler handler, IHybridWebView hybridWebView, object? arg)
```

Comment:
@PureWeen I think this needs that fakesync with the FireAndForget method - especially for logging?

This is a "pattern":

```cs
void MapSomething(IViewHandler handler, IView view, object args)
{
    MapSomethingAsync(handler, view, args).FireAndForget(handler);
}
Task MapSomethingAsync(IViewHandler handler, IView view, object args)
{
    // the real code
}
```

Reply:
This needs to be the FireAndForget system to make sure errors are logged and does not accidentally crash some iOS situations.